node/lib/statsd.js

```diff
@@ -0,0 +1,155 @@
+                clean(stat.tags.type);
+        case 'outbound.calls.operational-errors':
+            return prefix + '.' +
+                clean(stat.tags.service) + '.' +
+                clean(stat.tags['target-service']) + '.' +
+                clean(stat.tags['target-endpoint']) + '.' +
+                clean(stat.tags.type);
+        case 'outbound.calls.app-errors':
+            return prefix + '.' +
+                clean(stat.tags.service) + '.' +
+                clean(stat.tags['target-service']) + '.' +
+                clean(stat.tags['target-endpoint']) + '.' +
+                clean(stat.tags.type);
+        case 'outbound.calls.retries':
+            return prefix + '.' +
+                clean(stat.tags.service) + '.' +
+                clean(stat.tags['target-service']) + '.' +
+                clean(stat.tags['target-endpoint']) + '.' +
+                stat.tags['retry-count'];
+        case 'outbound.calls.latency':
+            return prefix + '.' +
+                clean(stat.tags.service) + '.' +
+                clean(stat.tags['target-service']) + '.' +
+                clean(stat.tags['target-endpoint']);
+        case 'outbound.calls.per-attempt-latency':
+            return prefix + '.' +
+                clean(stat.tags.service) + '.' +
+                clean(stat.tags['target-service']) + '.' +
+                clean(stat.tags['target-endpoint']) + '.' +
+                clean(stat.tags.peer) + '.' +
+                stat.tags['retry-count'];
+
+        // inbound
+        case 'inbound.calls.recvd':
+            return prefix + '.' +
+                clean(stat.tags['calling-service']) + '.' +
+                clean(stat.tags.service) + '.' +
+                clean(stat.tags.endpoint);
+        case 'inbound.calls.success':
+            return prefix + '.' +
+                clean(stat.tags['calling-service']) + '.' +
+                clean(stat.tags.service) + '.' +
+                clean(stat.tags.endpoint);
+        case 'inbound.calls.system-errors':
+            return prefix + '.' +
+                clean(stat.tags['calling-service']) + '.' +
+                clean(stat.tags.service) + '.' +
+                clean(stat.tags.endpoint) + '.' +
+                clean(stat.tags.type);
+        case 'inbound.calls.app-errors':
+            return prefix + '.' +
+                clean(stat.tags['calling-service']) + '.' +
+                clean(stat.tags.service) + '.' +
+                clean(stat.tags.endpoint) + '.' +
+                clean(stat.tags.type);
+        case 'inbound.calls.latency':
+            return prefix + '.' +
+                clean(stat.tags['calling-service']) + '.' +
+                clean(stat.tags.service) + '.' +
+                clean(stat.tags.endpoint);
+        default:
+            return 'tchannel.bad-stat-object';
+    }
+}
+
+TChannelStatsd.prototype.onStat = function onStat(stat) {
+    var self = this;
+    var key = getKey(self.channel.statTags, stat);
```

Comment:
dynamic dispatch is ok but you still want a default; just do whatever is readable.

I care more about a `switch` for `getKey()`